ReactiveCocoa/Objective-C/RACSignal.m

```diff
@@ -107,31 +107,36 @@ - (RACSignal *)bind:(RACStreamBindBlock (^)(void))block {
 	return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
 		RACStreamBindBlock bindingBlock = block();
 
-		NSMutableArray *signals = [NSMutableArray arrayWithObject:self];
+		__block NSUInteger signalsCount = 1;   // indicates self
```

Comment:
I think there will be a race condition between these two lines:

`OSAtomicDecrement32Barrier(&signalsCount);

if (signalsCount == 0) {`

We need an atomic 'decrement and test' operation.